resources/views/PhpReboot/Magazine/articleDetails.blade.php

```diff
@@ -76,11 +76,18 @@
                             <a href="#" data-toggle="modal" data-target="#myModal">
                                 Add to favorite (login needed)
                             </a>
+                        @else
+                            @if (isset($isFavourite) && $isFavourite)
+                                <a href="#" id="toggle-favourite">Remove from favourite</a>
+                            @else
+                                <a href="#" id="toggle-favourite">Add to favourite</a>
+                            @endif
+                        @endif
 
-                            <!-- Modal -->
-                            <div class="modal fade" id="myModal" tabindex="-1" role="dialog" aria-labelledby="myModalLabel">
-                                <div class="modal-dialog" role="document">
-                                    <div class="modal-content">
+                        <!-- Modal -->
```

Comment:
The modal window was once inside the if statement. 
So, when the session was lost (logout in another browser tab for example), it didn't show.

Now, when the session it's over, the jquery catches the "Not authorized" error message and shows modal asking for login/register.